### ISSUE TITLE
PhpdocAddMissingParamAnnotationFixer ignores comments that must be fixed

### ISSUE THREAD

**Issue by WinterSilence** [PHP-CS-Fixer/PHP-CS-Fixer#5508]
Class `PhpdocAddMissingParamAnnotationFixer`(ver. 2.18-3.0) have 2 not correct checks:

~~~php
if (false !== stripos($tokenContent, 'inheritdoc')) {
    continue;
}
// ignore one-line phpdocs like `/** foo */`, as there is no place to put new annotations
if (false === strpos($tokenContent, "\n")) {
    continue;
}
~~~

Why they not correct? Fixer MUST meet it's purpose - add parameter annotations. These checks should be optional and disabled by default.

PhpdocAddMissingParamAnnotationFixer miss fix:
~~~php
class Fix {
    /** lazy coder's comment  */
    function me(object $param) {
           return get_class($param);
    }
}
~~~
~~~php
class Fix2 extends Fix {
    /**
     * I forgot to add tag for new optional parameter :(
     *
     * @inheritDoc
     */
    function me(object $param, $newParam = false) {
          return parent::me($param) . ($newParam ? 'Extra' : 'Basic') ;
    }
}
~~~

**Comment by WinterSilence** [comment]
Also, PhpdocAddMissingParamAnnotationFixer generate invalid types like as `?int` or `null|mixed`, reformatted functions:
~~~php
  /**
   * this function is not real.
   * @param ?int $id
   */
  function test(?int $lookID = null)
  {
  }
  /**
   * this function is not real.
   * @param ?int $id
   */
  function test2(?int $id)
  {
  }
  /**
   * this function is not real.
   * @param null|mixed $id
   */
  function test3($id = null)
  {
  }
~~~

**Comment by WinterSilence** [comment]
Need more bugs for God of Bugs...

fix first founded docBlock:
~~~php
<?php
/**
 * don't fix me please
 * @param null|mixed $id
 */

/**
 * need fix
 */
function fixme($id = null)
{
}
~~~

**Comment by WinterSilence** [comment]
Fix some problems in PR #5509

**Comment by Wirone** [comment]
I reproduced it with `3.16`, but I'm going to close it because:
- there's no interest in this bug report (stale for 2 years)
- core team has limited capacity and won't work on it probably
- not critical from code's PoV
- native types should be promoted instead of phpDoc annotations

Feel free to provide proper PR anyway, with test cases and backward compatibility.
